packages/compiler/src/core/checker.ts

```diff
@@ -3461,6 +3462,59 @@ export function createChecker(program: Program, resolver: NameResolver): Checker
     }
 
     internalDecoratorValidation();
+
+    reportUnusedUsingAndImports();
+  }
+
+  function reportUnusedUsingAndImports() {
+    // Don't provide unused diagnostics if customer is using projection in the project because
+    // the projection statements will only be processed when applying projection. There is no way to determine
+    // whether "import" or "using" is referenced from them, so we just skip here to avoid providing incorrect suggestions (diagnostics)
+    // This should be fine for now considering projection is an experiemental feature.
+    for (const node of processedProjections) {
```

Comment:
@timotheeguerin , I tried to check the project part code and can't find a way to identify the references as commented here. Please let me know if there is some way I missed to count them considering my knowledge in compiler is still limited. thanks.